README.md

```diff
@@ -1 +1,679 @@
+
+We begin again with our 
+
+```js
+let test_data = `
+ELEMENT01=3
+ELEMENT04=4
+discarded:incorrect
+ELEMENT03=1
+foo=4
+element06=12
+ELEMENT07=12
+bar=3
+ELEMENT02=12
+`
+
+const readLines = (v) => v.split("\n")
+
+test_data
+    |> readLines(^)
+    |> console.log(^)
+
+```
+
+As we can see, the syntax is similar. Probably the most substantial difference is the `^` character. From a programming paradigm style, this changes the implementation from a [tacit](https://en.wikipedia.org/wiki/Tacit_programming) to a more explicit one. The character is the token we defined in the babel plugin; in other examples, you may see other characters being used. To my information, there is no decision yet on which character will be the default token.
+
+Running this code (after babel transforms it) will output the following result:
+
+```sh
+[
+  '',
+  'ELEMENT01=3',
+  'ELEMENT04=4',
+  'discarded:incorrect',
+  'ELEMENT03=1',
+  'foo=4',
+  'element06=12',
+  'ELEMENT07=12',
+  'bar=3',
+  'ELEMENT02=12',
+  ''
+]
+```
+
+Similar to f#, we got off with a good start. So, let's continue by tokenizing it. However, unlike in f#, we don't have Maybe or Option types at our disposal, so we are going to approach the error handling slightly different by filtering the elements that don't have an `=` first, and then splitting them: 
+
+```js
+const tokenize = (v) =>
+    v
+    |> ^.filter(v => v.indexOf("=") > -1)
+    |> ^.map(v => v.split("="))
+
+
+test_data
+    |> readLines(^)
+    |> tokenize(^)
+    |> console.log(^)
+```
+
+This is different from how we approached it in f#, which has the Option type allowing us to map and decide in one go. However, it's not too much of a change either, so we might be inclined to ignore it. Nevertheless, it is crucial to highlight a fundamental difference between the two approaches: In f#, we have a construct that allows us to specify the existence of a value and then bind the values to evaluate if we want to compute them. We consciously decided with the `Seq.choose` statement that we wanted to bind them right away. As an alternative, we could have also written:
+
+```fsharp
+    // before
+    v
+    |> readLines
+    |> Seq.map tokenize
+    |> Seq.choose each
+    |> Seq.filter (keyToUpper >> keyContains "ELEMENT")
+
+    // after
+    v
+    |> readLines
+    |> Seq.map tokenize
+    |> Seq.filter (fun (element) -> 
+        element
+        >>= fun ((key, _)) -> Some(key)
+        >>= toUpper
+        >>= contains "ELEMENT"
+        |> hasValue)
+```
+
+One may argue that the benefit of this approach is that it allows us to generalize the toUpper/contains functions away from the sequence format. But in this specific case, it makes the code more verbose without improving our sequence. Actually it reduces the quality, as before the `choose`-call gave our compiler a hint that no entry would be optional, whereas this might be the case in the second example. 
```

Comment:
```suggestion
One may argue that the benefit of this approach is that it allows us to generalize the toUpper/contains functions away from the sequence format. But in this specific case, it makes the code more verbose without improving our sequence. Actually, it reduces the quality, as before the `choose`-call gave our compiler a hint that no entry would be optional, whereas this might be the case in the second example. 
```